src/egui/mod.rs

```diff
@@ -453,29 +522,51 @@ impl UpdatesApp {
     
                     for pkg in update.packages.iter() {
                         // Avoid duplicates by checking if there's already a download for this serial and version on the queue.
-                        if !self.v.download_queue.iter().any(| d | &d.title_id == title_id && d.pkg_id == pkg.id()) {
+                        if self.get_active_download(&title_id, pkg).is_none() {
                             info!("Downloading update {} for serial {title_id} (group)", pkg.id());
-                            new_downloads.push(self.start_download(title_id.to_string(), title.clone(), pkg.clone()));
+                            self.add_download(self.start_download(title_id.to_string(), title.clone(), pkg.clone()));
                         }
                     }
                 }
+
+                if platform_variant != utils::PlaformVariant::PS4 { return; }
+
+                let is_multipart = update.packages.len() > 1;
+                let all_pkgs_completed = update.packages.iter().all(|pkg| {
+                    return self.pkg_download_status(title_id, pkg) == ActiveDownloadStatus::Completed;
+                });
+                let is_mergable = is_multipart && all_pkgs_completed;
+                let hover_text = if is_multipart {
+                    "All parts need to be completed for merge to be available"
+                } else {
+                    "This PS4 update is not a multipart update"
+                };
+                let merge_btn = ui.add_enabled(is_mergable, Button::new("Merge parts"))
```

Comment:
For consistency (and some personal preference) this should be
```suggestion
                let merge_btn = ui.add_enabled(is_mergable, egui::Button::new("Merge parts"))
```
which also renders the `use::eframe::egui::Button` redundant

Reply:
Done, added `egui` and removed `Button` import